controllers/core/flagd/resource.go

```diff
@@ -0,0 +1,74 @@
+package flagd
```

Comment:
+1 for having a common reconciler. can we rename the file to something like `resource_reconciler` so the idea is clear ?